components/permissions_gates/channel_permission_gate/index.js

```diff
@@ -8,7 +8,7 @@ import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles'
 import ChannelPermissionGate from './channel_permission_gate.jsx';
 
 function mapStateToProps(state, ownProps) {
-    if (!ownProps.teamId || !ownProps.channelId) {
+    if ((!ownProps.teamId && ownProps.teamId !== '') || !ownProps.channelId) {
```

Comment:
I am assuming teamId is empty only for DM and for GM here and if it doesn't exist then it is `null` or `undefined`. Waiting for someone to confim if my assuption is right

Reply:
the expression is "tricky", took me some time to understand the idea. We could change it to
`if (!ownProps.channelId || ownProps.teamId === null || ownProps.teamId === undefined) {`

Reply:
That looks better. sure I will make the change